.github/workflows/ci-on-comment.yaml

```diff
@@ -0,0 +1,123 @@
+name: CI
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  kata-containers-ci-on-comment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check for /test-gha
+        id: test-command
+        run: |
+          comment_body=$(
+          cat <<'END_COMMENT'
+          ${{ github.event.comment.body }}
+          END_COMMENT
+          )
+          if echo "$comment_body" | sed 's/\r$//' | grep -xq '^/test-gha$'; then
+            echo "is_present=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Fetch PR info
+        id: pr-info
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
On this PR's cover letter (which I loved!) you wrote `Because issue_comment workflows also don't have access to the repo secrets (...)`. So how does it has access to `secrets` here?

I couldn't find nothing about secrets on issue_comment doc: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment . But it says `This event will only trigger a workflow run if the workflow file is on the default branch` and other events that run on default branch (e.g. pull_request_target) have access to secrets...so perhaps it does as well but it is not documented?

Another possibility is that it works in your fork only. I've seen cases where this "event X hasn't access to secrets" rule simply does not apply when I run the workflow in a forked repository. 

I will pause my review here because the answers of those questions can change the whole thing.

Reply:
Hmm you do bring up a good point. I did some more digging here and I'm confused now.

The easy stuff first: from [1] and [2], `GITHUB_TOKEN` is a special secret that's always available (but it might not always have full permissions, although for this step we only need read permissions so we're good).

So for this step we're good anyway, since we're only reading. But for the `gh workflow run` step things get murky.

I'm not sure anymore whether `issue_comment` has access to secrets: it makes sense that `pull_request_review_comment` doesn't as that runs in the PR context, and I thought since `issue_comment` runs in the default branch context that it does have access (and nothing in the docs seem to indicate it can't access secrets).

However from [2] `With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.` - so whether it has access to secrets depends on whether `issue_comment` counts as "triggered from a forked repository".

Even then, the above isn't a problem as long as the underlying `ci.yaml` workflow has access to secrets. However, from [1] ("Permissions for the GITHUB_TOKEN") the maximum access for PRs from public forks for the `actions` scope is `read`. So again, this depends on whether `issue_comment` counts as public fork PR.

With all this, I'm not sure this is at all implementable. I suggest to add a manual permission override in the YAML and then merge this and see how far we get.

[1] https://docs.github.com/en/actions/security-guides/automatic-token-authentication
[2] https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions

Reply:
Added a `permissions` block to the job YAML, PTAL

Reply:
To conclude, as you mentioned, my hope is that `issue_comment` does allow us to call `gh workflow run`. Ideally even, `issue_comment` has access to all secrets and we can just use a `uses` block.